internal/clientconn/conn.go

```diff
@@ -581,26 +581,31 @@ func (c *conn) handleOpMsg(ctx context.Context, msg *wire.OpMsg, command string)
 //
 // The param `who` will be used in logs and should represent the type of the response,
 // for example "Response" or "Proxy Response".
-//
-// If response op code is not `OP_MSG`, it always logs as a debug.
-// For the `OP_MSG` code, the level depends on the type of error.
-// If there is no errors in the response, it will be logged as a debug.
-// If there is an error in the response, and connection is closed, it will be logged as an error.
-// If there is an error in the response, and connection is not closed, it will be logged as a warning.
 func (c *conn) logResponse(who string, resHeader *wire.MsgHeader, resBody wire.MsgBody, closeConn bool) zapcore.Level {
 	level := zap.DebugLevel
 
 	if resHeader.OpCode == wire.OpCodeMsg {
 		doc := must.NotFail(resBody.(*wire.OpMsg).Document())
 
-		ok, _ := doc.Get("ok")
-		if f, _ := ok.(float64); f != 1 {
-			if closeConn {
-				level = zap.ErrorLevel
-			} else {
-				level = zap.WarnLevel
-			}
+		var ok bool
+
+		v, _ := doc.Get("ok")
+		switch v := v.(type) {
+		case float64:
+			ok = v == 1
+		case int32:
+			ok = v == 1
+		case int64:
+			ok = v == 1
 		}
+
+		if !ok {
+			level = zap.WarnLevel
+		}
+	}
+
+	if closeConn {
+		level = zap.ErrorLevel
```

Comment:
Is it intended to change log level to errorLevel for `closeConn == true` even if `ok == 1`?

Reply:
I don't think that case exists: `closeConn` represents connection closing as a result of handling some command, and that's always something unrecoverable

Reply:
I see, so it would never be ok == 1 that case. Thanks for the explanation :hugs: